silo-core/contracts/lib/SiloMathLib.sol

```diff
@@ -143,19 +143,18 @@ library SiloMathLib {
         Math.Rounding _roundingToShares,
         ISilo.AssetType _assetType
     ) internal pure returns (uint256 assets, uint256 shares) {
-        require(_assets != 0 || _shares != 0, ISilo.InputZeroAssetsOrShares());
-
         if (_assets == 0) {
+            require(_shares != 0, ISilo.InputZeroShares());
             shares = _shares;
             assets = convertToAssets(_shares, _totalAssets, _totalShares, _roundingToAssets, _assetType);
+            require(assets != 0, ISilo.ReturnZeroAssets());
         } else if (_shares == 0) {
             shares = convertToShares(_assets, _totalAssets, _totalShares, _roundingToShares, _assetType);
```

Comment:
Should we not require require(_assets != 0, ISilo.ReturnZeroAssets());

Reply:
```
if (_assets == 0) {
  // we need shares > 0, we require that,
  // and then we require thet calculated asses > 0
} else if (_shares == 0) {
  // because we entered this block, asses are not 0, so we don't have to require that
  // we only need to check if calculated shares > 0
}
```